source/sigv4.c

```diff
@@ -1033,3 +1820,239 @@ SigV4Status_t SigV4_AwsIotDateToIso8601( const char * pDate,
 
     return returnStatus;
 }
+
+SigV4Status_t SigV4_GenerateHTTPAuthorization( const SigV4Parameters_t * pParams,
+                                               char * pAuthBuf,
+                                               size_t * authBufLen,
+                                               char ** pSignature,
+                                               size_t * signatureLen )
+{
+    SigV4Status_t returnStatus = SigV4Success;
+    CanonicalContext_t canonicalContext;
+    char * pPath = NULL, *pAlgorithm = NULL;
+    size_t pathLen = 0U, encodedLen = 0U, algorithmLen = 0U, authPrefixLen = 0U;
+    HmacContext_t hmacContext = { 0 };
+    SigV4String_t signingKey;
+
+    /*returnStatus = verifySigV4Parameters( pParams ); */
+
+    /* Default arguments. */
+    if( ( pParams->pAlgorithm == NULL ) || ( pParams->algorithmLen == 0 ) )
+    {
+        /* The default algorithm is AWS4-HMAC-SHA256. */
+        pAlgorithm = SIGV4_AWS4_HMAC_SHA256;
+        algorithmLen = SIGV4_AWS4_HMAC_SHA256_LENGTH;
+    }
+    else
+    {
+        pAlgorithm = pParams->pAlgorithm;
+        algorithmLen = pParams->algorithmLen;
+    }
+
+    /* Check if the authorization buffer has enough space to hold
+     * the final SigV4 Authorization header value. */
+    authPrefixLen = algorithmLen + SPACE_CHAR_LEN +                                            \
+                    AUTH_CREDENTIAL_PREFIX_LEN + pParams->pCredentials->accessKeyIdLen +           \
+                    CREDENTIAL_SCOPE_SEPARATOR_LEN + sizeNeededForCredentialScope( pParams ) + \
+                    AUTH_SEPARATOR_LEN + AUTH_SIGNED_HEADERS_PREFIX_LEN +                      \
+                    //sizeNeededFor
```

Comment:
sizeNeededFor what?

Reply:
Size needed for the signed headers as this needs to go after the signed headers are written to the processing buffer